src/fides/api/ops/schemas/dataset.py

```diff
@@ -1,219 +1,77 @@
-from typing import Any, Dict, List, Optional
+from typing import Any, List, Optional
 
-from fideslang.models import Dataset, DatasetCollection, DatasetFieldBase
-from pydantic import BaseModel, ConstrainedStr, Field, validator
+from fideslang.models import Dataset, DatasetCollection, DatasetField
+from fideslang.validation import FidesKey
+from pydantic import BaseModel, validator
+from sqlalchemy.orm import Session
 
-from fides.api.ops.common_exceptions import (
-    InvalidDataLengthValidationError,
-    InvalidDataTypeValidationError,
-)
-from fides.api.ops.graph.config import EdgeDirection
-from fides.api.ops.graph.data_type import is_valid_data_type, parse_data_type_string
+from fides.api.ctl.sql_models import DataCategory  # type: ignore[attr-defined]
+from fides.api.ops import common_exceptions
 from fides.api.ops.schemas.api import BulkResponse, BulkUpdateFailed
 from fides.api.ops.schemas.base_class import BaseSchema
-from fides.api.ops.schemas.shared_schemas import FidesOpsKey
-from fides.api.ops.util.data_category import _validate_data_category
+from fides.api.ops.util.data_category import (
+    DataCategory as DefaultTaxonomyDataCategories,
+)
 
 
-def _valid_data_categories(
-    data_categories: Optional[List[FidesOpsKey]],
-) -> Optional[List[FidesOpsKey]]:
+def validate_data_categories_against_db(dataset: Dataset, db: Session) -> None:
     """
-    Ensure that every data category provided matches a valid category defined in
-    the current taxonomy. Throws an error if any of the categories are invalid,
-    or otherwise returns the list of categories unchanged.
-    """
-
-    if data_categories:
-        return [dc for dc in data_categories if _validate_data_category(dc)]
-    return data_categories
+    Validate that data_categories defined on the Dataset, Collection, and Field levels exist
+    in the database.  Doing this instead of a traditional validator function to have
+    access to a database session.
 
+    If no data categories in the database, default to using data categories from the default taxonomy.
+    """
+    defined_data_categories: List[FidesKey] = [
+        cat[0] for cat in db.query(DataCategory.fides_key).all()
+    ]
+    if not defined_data_categories:
+        defined_data_categories = list(DefaultTaxonomyDataCategories.__members__.keys())
```

Comment:
Adding new validation on data category that looks in the database to see if data categories match ones defined there, instead of matching against the default data categories.

As a backup, if there's nothing in the db, which shouldn't be the case, matching against the static list.

Reply:
Would it make sense to add an `INFO` log saying what `DataCategory` didn't have any categories in the database? If it's a situation that shouldn't happen it might be worth making a note of it.